From fff3279bcf904673d312893b09bfc53811028490 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 28 May 2020 03:17:09 +0000
Subject: [PATCH] GCC: use brace-initializer for QuotaClientType base::flat_set

Constructing base::flat_set of QuotaClientType with parenthesis
constructor is ambiguous in GCC. Use brace-initializer to avoid
that problem.

Bug: 819294
Change-Id: Id33256a25d2af98d7be6ee570e2535a8b7bdf031
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2217554
Reviewed-by: Kinuko Yasuda <kinuko@chromium.org>
Commit-Queue: Kinuko Yasuda <kinuko@chromium.org>
Cr-Commit-Position: refs/heads/master@{#772542}
---

diff --git a/storage/browser/quota/quota_client_type.cc b/storage/browser/quota/quota_client_type.cc
index b9e9f62..a2daaa2 100644
--- a/storage/browser/quota/quota_client_type.cc
+++ b/storage/browser/quota/quota_client_type.cc
@@ -9,7 +9,7 @@
 namespace storage {
 
 const QuotaClientTypes& AllQuotaClientTypes() {
-  static base::NoDestructor<QuotaClientTypes> all({
+  static base::NoDestructor<QuotaClientTypes> all{{
       QuotaClientType::kFileSystem,
       QuotaClientType::kDatabase,
       QuotaClientType::kAppcache,
@@ -17,7 +17,7 @@
       QuotaClientType::kServiceWorkerCache,
       QuotaClientType::kServiceWorker,
       QuotaClientType::kBackgroundFetch,
-  });
+  }};
   return *all;
 }
 
